src/js/actions/policy.js

```diff
@@ -226,67 +219,84 @@ define(function (require, exports) {
             pointerPolicies: policyStore.getPolicies(PolicyStore.eventKind.POINTER)
         };
 
-        return adapterUI.getKeyboardPropagationMode()
-            .bind(this)
-            .then(function (mode) {
-                _cachedPolicyState.keyboardMode = mode;
-                return adapterUI.getPointerPropagationMode();
-            })
-            .then(function (mode) {
-                _cachedPolicyState.pointerMode = mode;
-                policyStore.clearPolicies(PolicyStore.eventKind.KEYBOARD);
-                policyStore.clearPolicies(PolicyStore.eventKind.POINTER);
-                return Promise.join(
-                    _syncPolicies.call(this, PolicyStore.eventKind.KEYBOARD),
-                    _syncPolicies.call(this, PolicyStore.eventKind.POINTER));
-            })
-            .then(function () {
-                return Promise.join(
-                    adapterUI.setKeyboardPropagationMode(
-                        { defaultMode: adapterUI.keyboardPropagationMode.FOCUS_PROPAGATE }),
-                    adapterUI.setPointerPropagationMode(
-                        { defaultMode: adapterUI.pointerPropagationMode.ALPHA_PROPAGATE }));
-            })
+        policyStore.clearPolicies(PolicyStore.eventKind.KEYBOARD);
+        policyStore.clearPolicies(PolicyStore.eventKind.POINTER);
+
+        var setKeyboardModePromise = adapterUI.getKeyboardPropagationMode()
+            .then(function (keyboardMode) {
+                _cachedPolicyState.keyboardMode = keyboardMode;
+
+                return adapterUI.setKeyboardPropagationMode(
+                    { defaultMode: adapterUI.keyboardPropagationMode.FOCUS_PROPAGATE });
+            });
+
+        var setPointerModePromise = adapterUI.getPointerPropagationMode()
+            .then(function (pointerMode) {
+                _cachedPolicyState.pointerMode = pointerMode;
+
+                return adapterUI.setPointerPropagationMode(
+                        { defaultMode: adapterUI.pointerPropagationMode.ALPHA_PROPAGATE });
+            });
+
+        var syncKeyboardPoliciesPromise = _syncPolicies.call(this, PolicyStore.eventKind.KEYBOARD),
+            syncPointerPoliciesPromise = _syncPolicies.call(this, PolicyStore.eventKind.POINTER);
+
+        return Promise.join(
```

Comment:
I see what you did here.  My original assumption was that there might be some interaction between setting policies, and the mode ... but I guess that's not the case, so OK!